src/parser/ast.c

```diff
@@ -291,14 +291,110 @@ AST_Validation _Validate_CREATE_Clause(const AST_QueryExpressionNode* ast, char
 }
 
 AST_Validation _Validate_SET_Clause(const AST_QueryExpressionNode* ast, char **reason) {
+	char undefined_alias[30] = "";
+	
+	if (NULL == ast->setNode) {
+		return AST_VALID;
+	}
+
+	if (!ast->matchNode) {
+		return AST_INVALID;
+	}
+	
+    Vector *update_aliases = NewVector(char*, Vector_Size(ast->setNode->set_elements));
+	for (int i=0; i < Vector_Size(ast->setNode->set_elements); i++) {
+		AST_SetElement *update_entity = NULL;
+        Vector_Get(ast->setNode->set_elements, i, &update_entity);
+		Vector_Push(update_aliases, update_entity->entity->alias);
+	}
+
+	AST_Validation res = _Validate_Aliases_In_Match_Clause(
+		update_aliases, ast->matchNode->graphEntities, undefined_alias);
+	Vector_Free(update_aliases);
+
+	if (res != AST_VALID) {
+		sprintf(*reason, "%s not defined", undefined_alias);
```

Comment:
reason is not allocated, you're writing to an unallocated memory,
use malloc to allocate before writing.